### PR TITLE
fix: ensure prefund setup is signed if turnNum < participants.length

### PR DESCRIPTION
**Pull request by snario**
I believe this is an off by one error. If your turn number is 0, and you have not signed anything yet, you want to sign the prefund setup if the latest turnNum is less than 2

### CODE REVIEW

**Review by andrewgordstewart — APPROVED** [review]
This should probably be accompanied with some unit tests to test the intended behaviour, in particular in the edge case you described.

**Comment by snario** [comment]
> This should probably be accompanied with some unit tests to test the intended behaviour, in particular in the edge case you described.

b27bb1b51

File: packages/server-wallet/src/protocols/__test__/open-channel.test.ts [hunk 1]
```diff
@@ -36,22 +42,25 @@ const fundChannelAction1 = fundChannel({
 const fundChannelAction2 = {...fundChannelAction1, expectedHeld: BN.from(5)};
 
 test.each`
-  supported        | latestSignedByMe | latest           | funding      | action                       | cond                                                                     | result
-  ${prefundState}  | ${prefundState}  | ${prefundState}  | ${funded}    | ${signState(postfundState2)} | ${'when the prefund state is supported, and the channel is funded'}      | ${'signs the postfund state'}
-  ${prefundState}  | ${prefundState}  | ${prefundState}  | ${notFunded} | ${fundChannelAction1}        | ${'when the prefund state is supported, and alice is first to deposit'}  | ${'submits transaction'}
-  ${prefundState1} | ${prefundState1} | ${prefundState1} | ${funded}    | ${fundChannelAction2}        | ${'when the prefund state is supported, and alice is secont to deposit'} | ${'submits transaction'}
+  supported           | latestSignedByMe    | latest              | funding      | action                         | cond                                                                     | result
```
Comment (andrewgordstewart) [inline packages/server-wallet/src/protocols/__test__/open-channel.test.ts:45]: As nice as it is to look at this table, statically, this is far too difficult to review :(

Maybe the `test.each(cases)` API is better for us. (It also means the cases can be typed, but I don't know if the test callback can take advantage of that.

Comment (andrewgordstewart) [reply packages/server-wallet/src/protocols/__test__/open-channel.test.ts:45]: Screen Shot 2020-10-27 at 9 41 43 AM

You do get typed cases!

Comment (andrewgordstewart) [reply packages/server-wallet/src/protocols/__test__/open-channel.test.ts:45]: (Let's table this discussion for another day...)

File: packages/server-wallet/src/protocols/__test__/open-channel.test.ts [hunk 2]
```diff
@@ -36,22 +42,25 @@ const fundChannelAction1 = fundChannel({
...
-  ${prefundState}   | ${postfundState2} | ${postfundState2} | ${funded}    | ${'when the prefund state is supported, and I have signed the postfund state'}
-  ${prefundState1}  | ${prefundState1}  | ${prefundState1}  | ${notFunded} | ${'when the prefund state is supported, and alice is second to deposit'}
-  ${postfundState2} | ${postfundState2} | ${postfundState2} | ${funded}    | ${'when the first postfund state is supported'}
+  supported           | latestSignedByMe    | latest              | funding      | cond
+  ${undefined}        | ${prefundState0}    | ${prefundState0}    | ${funded}    | ${'when I have signed the prefund state, but it is not supported'}
+  ${undefined}        | ${undefined}        | ${undefined}        | ${funded}    | ${'when there is no state'}
+  ${prefundState0}    | ${postfundState2}   | ${postfundState2}   | ${funded}    | ${'when the prefund state is supported, and I have signed the postfund state'}
+  ${altPrefundState0} | ${altPrefundState0} | ${altPrefundState0} | ${notFunded} | ${'when the prefund state is supported, and alice is second to deposit'}
```
Comment (andrewgordstewart) [inline packages/server-wallet/src/protocols/__test__/open-channel.test.ts:62]: Same comment as above.

**Review by andrewgordstewart — COMMENTED** [review]
This has already been merged, but I might as well leave my pending comments anyway.

File: packages/server-wallet/src/protocols/__test__/open-channel.test.ts [hunk 3]
```diff
@@ -36,22 +42,25 @@ const fundChannelAction1 = fundChannel({
...
+  ${undefined}        | ${undefined}        | ${reversedPFState1} | ${notFunded} | ${signState(reversedPFState0)} | ${'when the prefund state is not supported, and I am index 1'}           | ${'signs the prefund state'}
+  ${prefundState0}    | ${prefundState0}    | ${prefundState0}    | ${funded}    | ${signState(postfundState2)}   | ${'when the prefund state is supported, and the channel is funded'}      | ${'signs the prefund state'}
+  ${prefundState0}    | ${prefundState0}    | ${prefundState0}    | ${funded}    | ${signState(postfundState2)}   | ${'when the prefund state is supported, and the channel is funded'}      | ${'signs the postfund state'}
+  ${prefundState0}    | ${prefundState0}    | ${prefundState0}    | ${notFunded} | ${fundChannelAction1}          | ${'when the prefund state is supported, and alice is first to deposit'}  | ${'submits transaction'}
+  ${altPrefundState0} | ${altPrefundState0} | ${altPrefundState0} | ${funded}    | ${fundChannelAction2}          | ${'when the prefund state is supported, and alice is secont to deposit'} | ${'submits transaction'}
```
[the remark this answers is not in the document]
Comment (snario) [reply packages/server-wallet/src/protocols/__test__/open-channel.test.ts:51]: Alice is the first participant in `altPrefundState0`
